src/middleware/authMiddleware.ts

```diff
@@ -5,33 +5,114 @@ import { JwtPayload } from "../types/auth";
 import { User } from "../models/User";
 import { AppDataSource } from "../config/db";
 
-declare global {
-  namespace Express {
-    interface Request {
-      user?: JwtPayload;
-    }
-  }
-}
-
-export const authenticateToken = (
+export const authenticateToken = async (
   req: Request,
   res: Response,
   next: NextFunction
 ) => {
-  const authHeader = req.headers.authorization;
-  const token = authHeader && authHeader.split(" ")[1];
+
+  const authHeader = 
+    req.headers['authorization'] || 
+    req.headers['Authorization'] || 
+    req.headers['AUTHORIZATION'];
+
+  console.log(`Raw Authorization Header: ${authHeader}`);
+
+  if (!authHeader) {
+    console.error('No authorization header found');
+    return res.status(401).json({ 
+      message: "Unauthorized: No authorization header",
+      details: "Authorization header is missing" 
+    });
+  }
+
+  const normalizedHeader = authHeader.toString().trim();
+  
+  console.log(`Normalized Authorization Header: ${normalizedHeader}`);
+
+  const tokenMatch = normalizedHeader.match(/^Bearer\s+(.+)$/i);
+  
+  if (!tokenMatch) {
+    console.error('Invalid authorization header format');
+    return res.status(401).json({ 
+      message: "Unauthorized: Invalid authorization header format",
+      details: "Header must start with 'Bearer ' followed by token" 
+    });
+  }
+
+  const token = tokenMatch[1].trim();
 
   if (!token) {
```

Comment:
I will say once you get the `tokenMatch` go ahead and verify, if there is anything `verifyToken` will throw that error for you.

Reply:
I used the trim() to handle minor space-related errors, like unintended spaces in the token to ensure the token is clean and avoids common issues caused by extra whitespace in user input or external data

However, you are right, the verifyToken could do that by throwing errors for invalid tokens...

Noted...